name.abuchen.portfolio/src/name/abuchen/portfolio/math/Risk.java

```diff
@@ -0,0 +1,73 @@
+package name.abuchen.portfolio.math;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.DoubleStream;
+
+public final class Risk
+{
+
+    public static double calculateMaxDrawdown(long[] values)
+    {
+        Double peak = Double.MIN_VALUE;
+        List<Double> dd = new ArrayList<Double>();
+        Double maxDD = 0d;
+        int i = 0;
+        for (double value : values)
+        {
+            peak = Math.max(peak, value);
+            dd.add(i, (peak - values[i]) / peak);
+            maxDD = Math.max(maxDD, dd.get(i));
+            i++;
+        }
+        return maxDD;
+    }
+
+    private static double[] getReturns(long[] values)
+    {
+        double[] returns = new double[values.length - 1];
+        for (int i = 0; i < returns.length; i++)
+        {
+            returns[i] = (double) values[i + 1] / values[i];
+        }
+        return returns;
+    }
+
+    public static double calculateAverageVolatility(long[] values)
+    {
+        double[] vola = Risk.getReturns(values);
+        double average = DoubleStream.of(vola).average().getAsDouble();
+        double variance = 0d;
+        for (int i = 0; i < vola.length; i++)
+        {
+            variance = Math.pow(vola[i] - average, 2) + variance;
+        }
+        variance = variance / vola.length;
+        variance = Math.sqrt(variance);
+        return variance;
+    }
+
+    public static double calculateSemiVolatility(long[] values)
```

Comment:
Ich habe mein Wissen hierher:
https://www.wiso.uni-hamburg.de/fileadmin/bwl/statistikundoekonometrie/Kindermann/QUIAM_SoSe_2009/quiam_sitzung02_sose09n.pdf

In einem Satz geht es darum dass ein Ausschlag nach oben nicht schlimm ist, sondern nur solche Ausschläge betrachtet werden, die nach unten gehen (Downside Risikomaß) http://en.wikipedia.org/wiki/Downside_risk
Da ist auch die Semivarianz erwähnt.
Das war einfach ein quickwin, also habe ich es gemacht